src/modules/retrocompat-config.js

```diff
@@ -45,9 +48,10 @@ function envVars(config) {
  * @return {object}        Updated config object
  */
 module.exports = (config) => {
-  config = config || {};
-  envVars(config);
-  oldConfigNames(config);
+  let configuration = Object.assign({}, config);
+  configuration = configuration || {};
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Change this condition so that it does not always evaluate to "true". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.dialonce.io/coding_rules#rule_key=javascript%3AS2583)